### PR TITLE
Add functional test for 404 page

### PR DESCRIPTION
**Pull request by alexgibson**
@davehunt r?

### CODE REVIEW

**Comment by alexgibson** [comment]
Hmm... seems this gives the old legacy 404 page on prod and so the test fails... will investigate why.

File: tests/pages/page_not_found.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (davehunt) [inline tests/pages/page_not_found.py]: Nit: No need for `page` in the file name as it's in the `pages` directory.

File: tests/functional/test_page_not_found.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+
+from pages.page_not_found import NotFoundPage
+
+
+@pytest.mark.nondestructive
+def test_is_download_button_displayed(base_url, selenium):
```
Comment (davehunt) [inline tests/functional/test_page_not_found.py]: Should this be a smoke test?

**Comment by alexgibson** [comment]
I believe we may be blocked on adding a test for this page by [Bug 1008100](https://bugzilla.mozilla.org/show_bug.cgi?id=1008100). I could add a test that triggers a 404 page using a known URL that passes through to bedrock, such as `/foundation/foobar/`, but this won't pass locally however when in dev mode.

**Comment by davehunt** [comment]
If this is valuable coverage we could potentially make the path conditional based on environment, or mark the test as expected to fail based on environment. The best solution is getting the dependent bug fixed though. :smile:

**Comment by alexgibson** [comment]
Now that #3942 is merged and deployed, I believe this is now unblocked from merging. 

Quick r @davehunt?

**Comment by davehunt** [comment]
r+

**Pull request merged**